test/interactions/clickInteractionTests.ts

```diff
@@ -1,158 +1,208 @@
+          const endPoint = {x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isFalse(callback.getCalled(), "callback not called if mouse is released outside Component");
+        });
+
+        it("does not invoke callback if click is started outside Component for " + TestMethods.InteractionMode[mode], () => {
+          const startPoint = {x: SVG_WIDTH * 2, y: SVG_HEIGHT * 2};
+          const endPoint = {x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2};
+
+          clickPointWithMove(startPoint, endPoint, mode);
+
+          assert.isFalse(callback.getCalled(), "callback not called if click was started outside Component");
+        });
+
+        it("invokes callback if the mouse is moved out then back inside the Component before releasing for "
+          + TestMethods.InteractionMode[mode], () => {
+          TestMethods.triggerFakeInteractionEvent(mode,
+                                                  TestMethods.InteractionType.Start,
+                                                  component.content(),
+                                                  SVG_WIDTH / 2,
+                                                  SVG_HEIGHT / 2);
+          TestMethods.triggerFakeInteractionEvent(mode,
+                                                  TestMethods.InteractionType.Move,
+                                                  component.content(),
+                                                  SVG_WIDTH * 2,
+                                                  SVG_HEIGHT * 2);
+          TestMethods.triggerFakeInteractionEvent(mode,
+                                                  TestMethods.InteractionType.End,
+                                                  component.content(),
+                                                  SVG_WIDTH / 2,
+                                                  SVG_HEIGHT / 2);
+          assert.isTrue(callback.getCalled(),
+                        "callback still called if the mouse is moved out then back inside the Component before releasing");
+        });
+      });
 
+      it("does not trigger callback when touch event is cancelled", () => {
         TestMethods.triggerFakeTouchEvent("touchstart", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
         TestMethods.triggerFakeTouchEvent("touchcancel", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
         TestMethods.triggerFakeTouchEvent("touchend", component.content(), [{x: SVG_WIDTH / 2, y: SVG_HEIGHT / 2}]);
-        assert.isFalse(callbackCalled, "callback not called since click was interrupted");
-
-        svg.remove();
+        assert.isFalse(callback.getCalled(), "callback not called since click was interrupted");
       });
     });
   });
+
+  class TestClickCallback {
+    private called: boolean;
+    private lastPoint: Plottable.Point;
+
+    constructor() {
+      this.called = false;
+    }
+
+    public call(point?: Plottable.Point) {
```

Comment:
If the `point` is optional so that users of the class do not necessarily need to provide a point, would it make more sense to have a `default` point so that we do not have to `null` check?  Or are we ok if this is null?